src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/DisasterRecoveryService.cs

```diff
@@ -332,106 +342,122 @@ internal void PerformBackup(BackupOperation backupOperation)
         {
             backupOperation.PerformBackup();
         }
-        
+
         /// <summary>
-        /// Create a backup task for execution and cancellation
+        /// Create a backup script task
         /// </summary>
         /// <param name="sqlTask"></param>
         /// <returns></returns>
-        internal async Task<TaskResult> BackupTaskAsync(SqlTask sqlTask)
+        internal async Task<TaskResult> BackupScriptTaskAsync(SqlTask sqlTask)
```

Comment:
You're correct (apologies for lack of clarity). You can just have each step be a function call, no need for individual tasks in this case.